flocker/control/_config.py

```diff
@@ -1296,7 +1296,7 @@ def deployment_from_configuration(deployment_state, deployment_configuration,
             raise ConfigurationError("No known node with address {}.".format(
                 hostname))
         node = Node(uuid=node_states[hostname].uuid,
-                    applications=frozenset(node_applications),
+                    applications={app.name: app for app in node_applications},
```

Comment:
I'd forgotten about all this code...for parsing YAML configuration files supplied to the _compose endpoint by `flocker-deploy`
We should remove all that code, right?
Not in this branch, but it would be good housekeeping.